core/block_syncer_test.go

```diff
@@ -0,0 +1,206 @@
+	syncer := NewBlockSyncer(hclog.Default())
+	var emptyHash []byte
+
+	mockSyncerBlockHandler := BlockSyncerHandlerMock{}
+	defer syncer.Close()
+	if err := syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, emptyHash, &mockSyncerBlockHandler); err != nil {
+		t.Fatalf("got syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, emptyHash, mockSyncerBlockHandler) = %v; want = nil", err)
+	}
+}
+
+func TestSync(t *testing.T) {
+	syncer := NewBlockSyncer(hclog.Default())
+	existingPointHash, _ := hex.DecodeString(ExistingPointHashStr)
+
+	mockSyncerBlockHandler := BlockSyncerHandlerMock{}
+	defer syncer.Close()
+	if err := syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, existingPointHash, &mockSyncerBlockHandler); err != nil {
+		t.Fatalf("got syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, existingPointHash, mockSyncerBlockHandler) = %v; want = nil", err)
+	}
+}
+
+func TestSyncWithExistingConnection(t *testing.T) {
+	syncer := NewBlockSyncer(hclog.Default())
+	connection, _ := ouroboros.NewConnection(
+		ouroboros.WithNetworkMagic(NetworkMagic),
+		ouroboros.WithNodeToNode(true),
+		ouroboros.WithKeepAlive(true),
+	)
+	syncer.connection = connection
+	existingPointHash, _ := hex.DecodeString(ExistingPointHashStr)
+
+	mockSyncerBlockHandler := BlockSyncerHandlerMock{}
+	defer syncer.Close()
+	if err := syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, existingPointHash, &mockSyncerBlockHandler); err != nil {
+		t.Fatalf("got syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, existingPointHash, mockSyncerBlockHandler) = %v; want = nil", err)
+	}
+}
+
+func TestCloseWithConnectionNil(t *testing.T) {
+	syncer := NewBlockSyncer(nil)
+
+	if err := syncer.Close(); err != nil {
+		t.Fatalf("got syncer.Close() = %v; want = nil", err)
+	}
+}
+
+func TestCloseWithConnectionNotNil(t *testing.T) {
+	syncer := NewBlockSyncer(nil)
+	connection, _ := ouroboros.NewConnection(
+		ouroboros.WithNetworkMagic(NetworkMagic),
+		ouroboros.WithNodeToNode(true),
+		ouroboros.WithKeepAlive(true),
+	)
+	syncer.connection = connection
+
+	if err := syncer.Close(); err != nil {
+		t.Fatalf("got syncer.Close() = %v; want = nil", err)
+	}
+}
+
+func TestGetFullBlockWithConnectionNil(t *testing.T) {
+	syncer := NewBlockSyncer(hclog.Default())
+
+	var slot uint64
+	var hash []byte
+	if _, err := syncer.GetFullBlock(slot, hash); err == nil {
+		t.Fatalf("got syncer.GetFullBlock(slot, hash) = (_, %v); want = (_, no connection)", err)
+	}
+}
+
+func TestGetFullBlockWithConnectionNotNilExisting(t *testing.T) {
+	syncer := NewBlockSyncer(hclog.Default())
+	existingPointHash, _ := hex.DecodeString(ExistingPointHashStr)
+
+	mockSyncerBlockHandler := BlockSyncerHandlerMock{}
+	defer syncer.Close()
+	// .GetFullBlock panics without full initialize from .Sync
+	if err := syncer.Sync(NetworkMagic, NodeAddress, ExistingPointSlot, existingPointHash, &mockSyncerBlockHandler); err != nil {
+		panic("failed to start sync")
```

Comment:
Please use the https://pkg.go.dev/github.com/stretchr/testify/require and https://pkg.go.dev/github.com/stretchr/testify/assert packages for error checking, avoiding the use of panic. Additionally, it is advisable to use require or assert over `FatalF` for this context.